gluten-core/src/main/scala/org/apache/gluten/extension/columnar/heuristic/AddFallbackTags.scala

```diff
@@ -25,7 +25,10 @@ import org.apache.spark.sql.execution.SparkPlan
 // Add fallback tags when validator returns negative outcome.
 case class AddFallbackTags(validator: Validator) extends Rule[SparkPlan] {
   def apply(plan: SparkPlan): SparkPlan = {
-    plan.foreachUp { case p => addFallbackTag(p) }
+    plan.foreachUp {
+      case p if FallbackTags.maybeOffloadable(p) => addFallbackTag(p)
+      case _ =>
+    }
```

Comment:
Why this change is needed? Thanks.

Reply:
Those that have been tagged do not need to validate again , and #8580 issus1 can be resolved.

Reply:
I see. Could see whether [exclusive tag](https://github.com/apache/incubator-gluten/blob/0f4489aeccb691abc2f739b7d7a1f4cb0037c093/gluten-core/src/main/scala/org/apache/gluten/extension/columnar/FallbackTag.scala#L39-L43) can help?

Reply:
Exclusive tag could be added by `PushDownInputFileExpression` so validator doesn't add another tag.

Though I will be fine to both approaches.

Reply:
Do you have any other comments @zhztheplayer thank you